lib/friends/introvert.rb

```diff
@@ -658,6 +658,19 @@ def set_n_activities!(type)
       end
     end
 
+    def set_implicit_locations(_activities)
+      implicit_location = nil
+      @activities.reverse_each do |activity|
```

Comment:
Ah, using `reverse_each` is clever!

Reply:
Haha, this was a typo that happens to still work 😂